docs/architecture/adr-003-values-propagation.md

```diff
@@ -0,0 +1,416 @@
+# ADR 003: Propagation of Proposed Values
+
+## Changelog
+
+* 2025-03-18: Context and description of the problem
+* 2025-03-21: Current design description
+
```

Comment:
Adding a new entry here would be good.

Reply:
Suggestion? Past two day work was still on "Current design description"  😄